apps/frontend/src/app/5_pages/AavePage/components/BorrowPositionsList/components/EfficiencyModeCard/components/SwitchEModeForm/SwitchEModeForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+
+import {
+  Button,
+  ErrorBadge,
+  ErrorLevel,
+  Icon,
+  IconNames,
+  Paragraph,
+  Select,
+  SimpleTable,
+  SimpleTableRow,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { AmountRenderer } from '../../../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { config } from '../../../../../../../../../constants/aave';
+import { useAaveSetUserEMode } from '../../../../../../../../../hooks/aave/useAaveSetUserEMode';
+import { useAaveUserReservesData } from '../../../../../../../../../hooks/aave/useAaveUserReservesData';
+import { translations } from '../../../../../../../../../locales/i18n';
+import { EModeCategory } from '../../../../../../../../../types/aave';
+import { CollateralRatioHealthBar } from '../../../../../CollateralRatioHealthBar/CollateralRatioHealthBar';
+import { normalizeEModeSummary } from '../../EfficencyModeCard.utils';
+
+type SwitchEModeFormProps = {
+  current: EModeCategory;
+  categories: EModeCategory[];
+  onComplete: () => void;
+};
+
+export const SwitchEModeForm: FC<SwitchEModeFormProps> = ({
+  current,
+  categories,
+  onComplete,
+}) => {
+  const { handleSetUserEMode } = useAaveSetUserEMode();
+  const [category, setCategory] = useState<string>('');
+  const { summary, userReservesData, reservesData, timestamp } =
+    useAaveUserReservesData();
+
+  const categoriesOptions = useMemo(() => {
+    return categories
+      .filter(c => c.id !== current.id)
+      .map(category => ({
+        label: category.label,
+        value: String(category.id),
+      }));
+  }, [categories, current?.id]);
+
+  const selectedCategory = useMemo(() => {
+    return categories.find(c => c.id === Number(category));
+  }, [category, categories]);
+
+  const summaryAfterSwitch = useMemo(() => {
+    return normalizeEModeSummary(
+      selectedCategory?.id ?? 0,
+      reservesData,
+      userReservesData,
+      timestamp,
+    );
+  }, [userReservesData, reservesData, timestamp, selectedCategory?.id]);
+
+  const positionsInOtherCategories = useMemo(() => {
+    return summary.reserves.some(
+      r => r.reserve.eModeCategoryId !== Number(category) && r.borrowed.gt(0),
+    );
+  }, [summary, category]);
+
+  const confirmEnabled = useMemo(() => {
+    // cannot switch if undercollateralized or have positions in other categories
+    return (
+      !summaryAfterSwitch.liquidationRisk &&
+      !positionsInOtherCategories &&
+      selectedCategory
+    );
+  }, [summaryAfterSwitch, positionsInOtherCategories, selectedCategory]);
+
+  const onConfirm = useCallback(() => {
+    if (!selectedCategory) return;
+    handleSetUserEMode(selectedCategory, { onComplete });
+  }, [handleSetUserEMode, selectedCategory, onComplete]);
```

Comment:
shouldn't be disable the button when there is no category? I'm thinking the scenary in which the user click the button without receiving any feedback.

Reply:
Indeed it is, this for typing porpouses, can aswell use `!` if you think it's cleaner